### PR TITLE
Feat/dapp 544

### PR DESCRIPTION
**Pull request by leodhb**
# Description
- use a single payment page for everything
- create a theme change engine to change the color theme by current flow (cause/non profit)
- use a single donation done page to all the credit card flows

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

Put here all the things need to test this PR and verify your changes, also list any relevant details for tests (eg: have a wallet)

- Test A

### CODE REVIEW

File: src/pages/promoters/PaymentPage/styles.ts [hunk 1]
```diff
@@ -81,10 +81,12 @@ export const DonateButtonContainer = styled.div`
   }
 `;
 
-export const DonateButton = styled(Button)`
-  border-color: ${({ theme }) => theme.colors.orange20};
-  background-color: ${({ theme }) => theme.colors.orange20};
-  color: ${({ theme }) => theme.colors.orange40};
+export const DonateButton = styled(Button)<{
+  colorTheme: any;
```
Comment (letbueno) [inline src/pages/promoters/PaymentPage/styles.ts:85]: here too

**Review by letbueno — APPROVED** [review]

File: src/pages/promoters/PaymentPage/UserInfoSection/styles.tsx [hunk 2]
```diff
@@ -2,13 +2,20 @@ import styled from "styled-components";
 import InputAutoComplete from "components/atomics/inputs/InputAutoComplete";
 import InputText from "components/atomics/inputs/InputText";
 
-export const BillingInformationSectionContainer = styled.div`
+export const BillingInformationSectionContainer = styled.div<{
+  colorTheme: any;
```
[the remark this answers is not in the document]
Comment (leodhb) [reply src/pages/promoters/PaymentPage/UserInfoSection/styles.tsx:6]: Actually it receives a object from _themeByFlow_, https://github.com/RibonDAO/interface/pull/729/files#diff-8a6ae304424b863d2bf3ac2c036d04f10ab02e165a297d40f4cf6d574d6a3217

[the remark this answers is not in the document]
Comment (leodhb) [reply src/pages/promoters/PaymentPage/UserInfoSection/styles.tsx:6]: Maybe we can specify a bit more what kind of object it is. I will do it! Nice catch, @letbueno <3

File: src/pages/promoters/PaymentPage/styles.ts (hunk 1 above)
Comment (leodhb) [reply src/pages/promoters/PaymentPage/styles.ts:85]: Ditto

**Pull request merged**